champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/assassin/sword/Concussion.java

```diff
@@ -89,13 +84,35 @@ public void onDamage(CustomDamageEvent e) {
         if (level <= 0) return;
 
         if (active.contains(damager.getUniqueId())) {
+            if (championsManager.getEffects().hasEffect(damagee, EffectTypes.CONCUSS)) {
+                UtilMessage.simpleMessage(damager, "<alt>" + damagee.getName() + "</alt> is already concussed.");
+                return;
+            }
+
             e.addReason("Concussion");
-            championsManager.getEffects().addEffect(damagee, damager, EffectTypes.BLINDNESS, 1, (long) (getDuration(level) * 1000));
+            championsManager.getEffects().addEffect(damagee, damager, EffectTypes.CONCUSS, 1, (long) (getDuration(level) * 1000L));
+
             UtilMessage.simpleMessage(damager, getName(), "You gave <alt>" + damagee.getName() + "</alt> a concussion.");
             UtilMessage.simpleMessage(damagee, getName(), "<alt>" + damager.getName() + "</alt> gave you a concussion.");
             active.remove(damager.getUniqueId());
         }
+    }
+
+    @EventHandler
+    public void hittingWhileConcussed(CustomDamageEvent e) {
+        if (e.getCause() != DamageCause.ENTITY_ATTACK) return;
+        if (!(e.getDamager() instanceof Player damager)) return;
+        if (!championsManager.getEffects().hasEffect(damager, EffectTypes.CONCUSS)) return;
+
+        UUID damagerUUID = damager.getUniqueId();
+
+        if (!previouslyHitWhileConcussed.contains(damagerUUID)) {
```

Comment:
This wont work in a real environment, what if the damager hits 2 seperate players? The Set you have created is global, you might want to look to create a WeakHashMap instead